storagedriver/wrapper.go

```diff
@@ -0,0 +1,100 @@
+package storagedriver
+
+import (
+	"io"
+)
+
+// Wrapper wraps the underlying storage driver with path checks
+// for each method.
+type Wrapper struct {
+	driver StorageDriver
+}
+
+// Wrap creates a wrapper for given storage driver to validate given paths
```

Comment:
This comment should probably indicate that `Wrap` implements common operations against a `StorageDriver`, rather than just path checks. For example, it might provide default implementations or error validation.